homeassistant/components/melnor/switch.py

```diff
@@ -21,52 +43,64 @@ async def async_setup_entry(
     async_add_devices: AddEntitiesCallback,
 ) -> None:
     """Set up the switch platform."""
-    switches = []
+    entities: list[MelnorZoneSwitch] = []
 
     coordinator: MelnorDataUpdateCoordinator = hass.data[DOMAIN][config_entry.entry_id]
 
     # This device may not have 4 valves total, but the library will only expose the right number of valves
     for i in range(1, 5):
-        if coordinator.data[f"zone{i}"] is not None:
-            switches.append(MelnorSwitch(coordinator, i))
-
-    async_add_devices(switches)
-
-
-class MelnorSwitch(MelnorBluetoothBaseEntity, SwitchEntity):
+        valve = coordinator.data[f"zone{i}"]
+        if valve is not None:
+
+            entities.append(
+                MelnorZoneSwitch(
+                    coordinator,
+                    valve,
+                    MelnorSwitchEntityDescription(
```

Comment:
Do you plan on adding more switch entities? 

The entity description is nice, but adds complexity if the integration only has one set of switches since you have another level of indirection with the lambdas.

Reply:
There will be several additional switches in the future. I need to plumb the functionality through the library yet before I can bring them in here